src/components/ButtonWithDropdownMenu.js

```diff
@@ -0,0 +1,139 @@
+import React, {useState, useRef, useEffect} from 'react';
+import PropTypes from 'prop-types';
+import {View} from 'react-native';
+import _ from 'underscore';
+import useWindowDimensions from '../hooks/useWindowDimensions';
+import styles from '../styles/styles';
+import Button from './Button';
+import PopoverMenu from './PopoverMenu';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import themeColors from '../styles/themes/default';
+import CONST from '../CONST';
+
+const propTypes = {
+    /** Text to display for the menu header */
+    menuHeaderText: PropTypes.string,
+
+    /** Callback to execute when the main button is pressed */
+    onPress: PropTypes.func.isRequired,
+
+    /** Whether we should show a loading state for the main button */
+    isLoading: PropTypes.bool,
+
+    /** Should the confirmation button be disabled? */
+    isDisabled: PropTypes.bool,
+
+    /** Additional styles to add to the component */
+    style: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.object), PropTypes.object]),
+
+    /** Menu options to display */
+    /** e.g. [{text: 'Pay with Expensify', icon: Wallet}, {text: 'PayPal', icon: PayPal}] */
+    options: PropTypes.arrayOf(
+        PropTypes.shape({
+            value: PropTypes.string.isRequired,
+            text: PropTypes.string.isRequired,
+            icon: PropTypes.elementType,
+            iconWidth: PropTypes.number,
+            iconHeight: PropTypes.number,
+            iconDescription: PropTypes.string,
+        }),
+    ).isRequired,
+};
+
+const defaultProps = {
+    isLoading: false,
+    isDisabled: false,
+    menuHeaderText: '',
+    style: [],
+};
+
+const ButtonWithDropdownMenu = (props) => {
+    const [selectedItemIndex, setSelectedItemIndex] = useState(0);
+    const [isMenuVisible, setIsMenuVisible] = useState(false);
+    const [popoverAnchorPosition, setPopoverAnchorPosition] = useState(null);
+    const {width: windowWidth, height: windowHeight} = useWindowDimensions();
+    const caretButton = useRef(null);
+    useEffect(() => {
+        if (!caretButton.current) {
+            return;
+        }
+        caretButton.current.measureInWindow((x, y, w, h) => {
+            setPopoverAnchorPosition({
+                horizontal: x + w,
+                vertical: y + h,
+            });
+        });
+    }, [windowWidth, windowHeight]);
+
```

Comment:
This caused a regression https://github.com/Expensify/App/issues/20262. The popover position is calculated too early at the first render where animation is not done yet (position is not stable yet).